DatadogSessionReplay/Sources/Recorder/ViewTreeSnapshotProducer/ViewTreeSnapshot/NodeRecorders/UITabBarRecorder.swift

```diff
@@ -7,21 +7,71 @@
 #if os(iOS)
 import UIKit
 
-internal struct UITabBarRecorder: NodeRecorder {
+internal final class UITabBarRecorder: NodeRecorder {
     let identifier = UUID()
+
+    private var currentlyProcessedTabbar: UITabBar? = nil
+    private lazy var subtreeViewRecorder: ViewTreeRecorder = {
```

Comment:
/change-request The `UITabBarRecorder` is static and the `currentlyProcessedTabbar ` var will retain the last recorded `UITabBar`. We should release the tab bar when we are done recording, for that we could use a method that instantiate a `ViewTreeRecorder` on the fly:
```swift
func recordSubtree(of tabbar: UITabBar, in context: ViewTreeRecordingContext) -> NodeSemantics? {
  ViewTreeRecorder([...]).record(tabBar, in: context)
}
```
I prefer this approach instead of a weak var, the flow is clearer this way IMO.

Reply:
That's a valid point, I made the requested change. Thanks for the suggestion!